### ISSUE TITLE
LocalizedLine's Text property setter is internal

### ISSUE THREAD

**Issue by McJones** [YarnSpinnerTool/YarnSpinner-Unity#120]
**What is the current behavior?**

The current flow of YS is that line providers create a custom localised line that is going to be used by the views, this line is passed over to the dialogue runner who passes it on up to the various views.

As part of this passing the runner adds the Text property onto the localised line before passing it along.
While this works it does feel a little weird that a core component of the line isn't set by the object responsible for making them, however if you do want to move this into line providers you also can't easily do this.

The Text property of a LocalizedLine setter is declared internal instead of public preventing modification.
In the case of the provided line providers this is fine, but if you need to make your own a core piece of a line is restricted from you.
Additionally if you do need to modify the Text property for the needs of your game you also can't do that.

In summary its weird that:
- line providers don't really provide lines
- a core component of the line can't be modified

**Please tell us about your environment:**
  
  - Yarn Spinner Version: v2.0.0-beta4
  - Unity Version: 2020.3.16f

**Other information** 

I think this can be fixed either by allowing Text to be modified and move the markup side of things into the line providers or change the localised line constructor to take in a markup object that can be given to the line provider by the dialogue runner when the runner requests a line.
